lib/os/cbprintf_packaged.c

```diff
@@ -480,8 +494,15 @@ int cbvprintf_package(void *packaged, size_t len, uint32_t flags,
 
 			bool is_ro = (fros_cnt-- > 0) ? true : ptr_in_rodata(s);
 			bool do_ro = !!(flags & CBPRINTF_PACKAGE_ADD_RO_STR_POS);
+#if defined(CONFIG_LOG2_FMT_SECTION)
+			bool is_string = ptr_in_string_section(s);
```

Comment:
Since ifdefs from function is removed, it can be simplified to:
```
bool is_string = IS_ENABLED(CONFIG_LOG2_FMT_SECTION) ? ptr_in_string_section(s) : false;
```